openid-4-verifiable-presentations-1_0.md

```diff
@@ -549,11 +555,276 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
 
 If the Verifier responds with any HTTP error response, the Wallet MUST terminate the process.
 
+# Verifiable Presentation Query Language {#vp_query}
```

Comment:
We had a discussion on optional data elements, documents etc. We were thinking of it conceptually like this:

A request for credentials aims to solve one or more (combination of) use cases. When addressing a particular use case the goal of the query language is to be have no ambiguity on what response addresses the use case or not. 

When addressing a particular use-case there should therefore not be any optional elements or documents, since if the use case is addressed with a particular combination of documents with their data elements providing more data elements is not necessary. 

But there can be requests that can address more than one use case. In that case there can be reasons to have some form of user choice. By having the use case layer to be the only layer where there is a form of optionality, it makes wallet implementations more straightforward and it also puts the place of optionality (which may mean user choice) at the place where it's most meaningful to the user (i.e. do you want to address use case x and/or y).

As a side-note If and when we have a use-case/purpose indicator, that's then also the place where we should do it.

Reply:
this has been discussed in the WG and has been addressed in the latest commits